### PR TITLE
irmin-pack: remove code duplication in Indexable and Atomic_write

### PR DESCRIPTION
**Pull request by metanivek**
- Address a couple of `FIXME` comments by removing code duplication in `irmin-pack`
- Update code for consistency after refactor

### CODE REVIEW

File: src/irmin/atomic_write.ml [hunk 1]
```diff
@@ -17,65 +17,45 @@
...
+  type value = AW.value
+  type watch = AW.watch
 
-  let check_not_closed t = if !(t.closed) then raise Store_properties.Closed
+  let make_closeable t = { closed = ref false; t }
 
-  let mem t k =
-    check_not_closed t;
-    S.mem t.t k
+  let get_open_exn t =
```
Comment (zshipko) [inline src/irmin/atomic_write.ml:28]: `get_open_exn` is a little confusing, maybe `store_if_open_exn` or `get_if_open_exn` would be more clear?

**Review by zshipko — COMMENTED** [review]
Looks like there is a CI issue, but otherwise this seems good!

File: src/irmin/atomic_write.ml (hunk 1 above)
Comment (metanivek) [reply src/irmin/atomic_write.ml:28]: Yeah, I just copied the existing function but I think `get_if_open_exn` is a better name. I'll make this change. :+1:

**Comment by metanivek** [comment]
CI is passing now. I had to change the code slightly to make older versions of OCaml happy -- the labeled argument has to be inside the paren on this line: https://github.com/mirage/irmin/pull/2045/files#diff-1b3ca33a8c8d857cfa49b8928ea68c6363c68de80c74a0a9e3e28c72a8bc87dcR32.

**Pull request merged**